x/dex/migrations/v2_to_v3.go

```diff
@@ -0,0 +1,212 @@
+package migrations
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/sei-protocol/sei-chain/x/dex/keeper"
+	"github.com/sei-protocol/sei-chain/x/dex/types"
+	legacytypes "github.com/sei-protocol/sei-chain/x/dex/types/legacy/v0"
+)
+
+const CONTRACT_ADDRESS_LENGTH = 62
+
+func DataTypeUpdate(ctx sdk.Context, storeKey sdk.StoreKey, cdc codec.BinaryCodec) error {
+	MigrateLongBooks(ctx, storeKey, cdc)
+	MigrateShortBooks(ctx, storeKey, cdc)
+	MigrateSettlements(ctx, storeKey, cdc)
+	MigrateTwap(ctx, storeKey, cdc)
+	MigrateRegisteredPairs(ctx, storeKey, cdc)
+	return nil
+}
+
+func MigrateLongBooks(ctx sdk.Context, storeKey sdk.StoreKey, cdc codec.BinaryCodec) error {
+	longBookKey := types.KeyPrefix(types.LongBookKey)
+	store := prefix.NewStore(ctx.KVStore(storeKey), longBookKey)
+	iterator := sdk.KVStorePrefixIterator(store, []byte{})
+
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		var val legacytypes.LongBook
+		cdc.MustUnmarshal(iterator.Value(), &val)
+		key := iterator.Key()
+		store.Delete(key)
+		priceDenom, err := types.GetDenomFromStr(val.Entry.PriceDenom)
+		if err != nil {
```

Comment:
rather than just continuing, should we log it so that we can revisit and try to fix this later? If we delete at the end the value is still there

Reply:
together with the `.Delete` comment above, my thought was to prevent any bad state from lingering in the blockchain after the migration since a bad state could potentially crash the chain during logics in Begin/EndBlock, and I don't know of a good way to fix bad states after migrations. Maybe we can just panic here? So that migration will fail, and we can go back and add in any missed edge cases and rerun migration